### ISSUE TITLE
XATMI api service flags not according to spec

### ISSUE THREAD

**Issue by albinsjolund** [casualcore/casual#14]
Either to 1.5.x  or 1.6.0

**Comment by OlofGullnas** [comment]
What is wrong? Is this an issue corresponding to issue #206 in bitbucket (a closed issue)? I think that the flags now are correct. At least on service invocation. Most of this was fixed when conversational was fixed. The only missing part is unit tests for this. There are some rudimentary tests as part of the unit tests for conversational, as conversational uses/relies on some of the flags.

**Comment by OlofGullnas** [comment]
I looked at this. The possible/allowed flags on service invocation are TPCONV, TPTRAN, TPNOREPLY, TPSENDONLY and TPRECVONLY.
As part of the refactoring after making conversational work Fredrik introduced code in common/source/server/handle/service.cpp to control the flags. The code is in common::server::handle::service::transform. This replaced the "hacky" setting of the flags I introduced while working on the conversational calls.

- TPTRAN (in_transaction) is set if transaction::context().current() is true. This is done for booth conversational and tpcall/tpacall to a service. It is the transaction context in the server that controls this flag. If the caller includes a transaction or not in the tpcall or tpconnect influences but do not control the flag. Service settings/atributes (Transaction : auto, transaction: join, ...) also influence the transaction state on invocation.  Correct.
- TPCONV (conversation) is set for conversational service invocations, and only for those. Correct.
- one of TPSENDONLY/TPRECVONLY (send_only/Flag::receive_only) is set depending on duplex in the connect request message. Can only be set for conversational services. Correct.  
- TPNOREPLY (no_reply) is set if a message::service::call::callee::Request has the no_reply flag set in its flags field. TPNOREPLY 
 can not be set for conversational. This is also correct.

The conclusion is that the flags are being set correctly.

There are some rudimentary unit tests for this as part of the conversational tests, but none with tpcall/tpacall. Some of the conversational tests return the invocation flags in a response to the "connector" that then verifies them. A similar strategy could be introduced for verifying the flags on tpcall/tpacall service invocation (an echo_service_flags service,,,).  Obviously except for TPNOREPLY. There are only two flags to verify on normal service invocations (TPTRAN and TPNOREPLY) so there are not much to verify.
